test/support/sugar_client.coffee

```diff
@@ -25,10 +25,12 @@ class SugarClient
     @client.id (id) => deferred.resolve @primus.spark id
     deferred.promise
   
-  ping: ->
+  pong: ->
     deferred = Bluebird.defer()
-    @client.write "primus::ping::#{ +new Date() }"
-    @client.once 'incoming::pong', -> deferred.resolve()
+    @client.write "primus::pong::#{ +new Date() }"
+    setTimeout ->
```

Comment:
I think this can use the private client method `incoming::ping` https://github.com/primus/primus#events instead of the timeout to allow the promise to resolve. 

maybe you tried this already?

Reply:
There’s something like a 10s wait until the server sends the next ping, so the tests would time out before that resolves.

Reply:
No, 10s is the ping timeout in the client. I’m not sure how frequently the server pings the client. 

This test does time out if we wait for a ping before resolving. 

Here we want to resolve once the server has handled the client pong, if we want to keep the tests as they are currently written. 

It might be better to rewrite the tests to send a server ping then run the tests after the expected client pong is received and handled.